File: stobuyonetran.py
class stobuyonetran:
    @staticmethod
    def maxProfit(self, prices):
        initial_Price = res = 0
        for price in prices:
            initial_Price = min(initial_Price, price)
            res  = max(res, price - initial_Price)
        return res
if __name__ == "__main__":
    prices = [7, 10, 1, 3, 6, 9, 2]
    print(stobuyonetran(prices))

File: stobuyonetran.cpp
#include<bits/stdc++.h>
using namespace std;
class stobuyonetran{
public:
    static int maxProfit(vector<int> &prices) {
        int initialPrices = prices[0], res = 0;
        for (int i = 0; i < prices.size(); i++)
        {
            initialPrices = min(initialPrices, prices[i]);
            res = max(res, prices[i] - initialPrices);
        }
        return res;
    }
};
int main() {
    vector<int> prices = {7, 10, 1, 3, 6, 9, 2};
    cout << stobuyonetran::maxProfit(prices) << endl;
    return 0;
}